README.md

```diff
@@ -1,61 +1,138 @@
+RUN circleci-install ruby 2.2.1
+RUN circleci-install scala
+
+# You can add custom files
+ADD my-custom-root-ca.crt /usr/local/share/ca-certificates/my-custom-root-ca.crt
+RUN update-ca-certificates
 ```
 
-Testing on a `m4.large`, building a container takes ~15-20 minutes.
+And then you can build it as you would typically do (i.e. `docker build -t my-container:0.3 .`).
+
+With this workflow, you will be basing your container on CircleCI published container image,
+speeding up creation process.  When CircleCI publishes new container image, you can rebuild
+your image and Docker will pick up the latest published version.
+
+**TODO**: Document available base image and document them
+
+## Building a container with substantial changes
+
+If you need to customize the image by starting from a clean slate, and/or
+require significant changes, e.g. different MySQL version, or MariaDB rather
+than MySQL, then the recommended path is to fork the repo and apply your
+changes, then build the Docker image.
+
+Please note, that this repo uses `circleci/ubuntu-server` as the base image.
+It uses the official Ubuntu Server image (rather than Ubuntu Core, the default
+ubuntu in DockerHub), tweaks slightly so Upstart can work with Docker.
+
+It's recommended that you fork this repo rather than start from scratch for the following benefits:
+
+* Ensure that CircleCI specific required customizations (e.g. xvfb) is correctly applied
+* Have a mechanism to pick up later tweaks and improvements by merging the repo - rather than merging them manually later
+* Reuse the custom provisioning framework that we are building to ease installation
+* Have a mechanism to push your changes/tweaks back to us
 
-*TODO* Better document the aws parameters
+## Super advanced mode: Using Chef/Ansible/etc
 
+If you have a significant infrastructure using custom provisioners, e.g.
+Chef/Ansible/SaltStack, please reach out to us private.  The exact instructions
+are beyond the scope of this documentation.
 
-## How to tweak
+In a very high level, we would recommend using [Packer](https://www.packer.io)
+and [Packer's Docker Builder](https://www.packer.io/docs/builders/docker.html).
 
-The installation scripts are found in `scripts/` directory.  Python/Ruby/NodeJS
-files allow ability to install multiple versions - you can edit the file to
-install the versions you need.
+# 2. Testing the container
 
-You can add new modules for frameworks/tools you use.  You can activate the module and set the execution order by inserting it into `build.sh` `MODULES` array.
+So you built the container successfully!  Congrats.  How can you test it?
 
+You can use the common Docker techniques for launching a container and connecting to it.
 
-## General Repo TODOs:
+The workflow, I adopt is the following:
+
+```bash
+host $ # build the image and name it `example-image`
+host $ docker build -t example-image .
+[...]
+host $ # Start it - but enable ssh and mount a sample test project
+host $ docker run -d \
```

Comment:
For some purposes yes.  With using ssh, you also get to verify that all shell/user profile customizations get executed.  In the last, I've seen some applications that behave slightly different when running with `sudo` and such.  Additionally, it also helps verify that sshd service is configured properly, since that's what CircleCI is going to use.